src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -2225,103 +2241,127 @@ def update_azure_service_mesh_profile(self) -> ServiceMeshProfile:
                     "for more details on enabling Azure Service Mesh."
                 )
+            new_profile.mode = CONST_AZURE_SERVICE_MESH_MODE_DISABLED
+            updated = True
+        elif enable_asm:
+            if new_profile is not None and new_profile.mode == CONST_AZURE_SERVICE_MESH_MODE_ISTIO:
+                raise ArgumentUsageError(
+                    "Istio has already been enabled for this cluster, please refer to "
+                    "https://aka.ms/asm-aks-upgrade-docs for more details on updating the mesh profile."
+                )
             requested_revision = self.raw_param.get("revision", None)
-            if mesh_upgrade_command in (
-                CONST_AZURE_SERVICE_MESH_UPGRADE_COMMAND_COMPLETE,
-                CONST_AZURE_SERVICE_MESH_UPGRADE_COMMAND_ROLLBACK,
-            ):
-                if len(new_profile.istio.revisions) < 2:
-                    raise ArgumentUsageError('Azure Service Mesh upgrade is not in progress.')
+            new_profile.mode = CONST_AZURE_SERVICE_MESH_MODE_ISTIO
+            if new_profile.istio is None:
+                new_profile.istio = self.models.IstioServiceMesh()  # pylint: disable=no-member
+            if mesh_upgrade_command is None and requested_revision is not None:
+                new_profile.istio.revisions = [requested_revision]
+            updated = True
 
-                sorted_revisons = self._sort_revisions(new_profile.istio.revisions)
-                if mesh_upgrade_command == CONST_AZURE_SERVICE_MESH_UPGRADE_COMMAND_COMPLETE:
-                    revision_to_remove = sorted_revisons[0]
-                    revision_to_keep = sorted_revisons[-1]
-                else:
-                    revision_to_remove = sorted_revisons[-1]
-                    revision_to_keep = sorted_revisons[0]
-                msg = (
-                    f"This operation will remove Istio control plane for revision {revision_to_remove}. "
-                    f"Please ensure all data plane workloads have been rolled over to revision {revision_to_keep} "
-                    "so that they are still part of the mesh.\nAre you sure you want to proceed?"
-                )
-                if prompt_y_n(msg, default="y"):
-                    new_profile.istio.revisions.remove(revision_to_remove)
-                    updated = True
-            elif (
-                mesh_upgrade_command == CONST_AZURE_SERVICE_MESH_UPGRADE_COMMAND_START and
-                requested_revision is not None
-            ):
-                if new_profile.istio.revisions is None:
-                    new_profile.istio.revisions = []
-                new_profile.istio.revisions.append(requested_revision)
-                updated = True
+        return new_profile, updated
+
+    # pylint: disable=too-many-branches,too-many-locals,too-many-statements
+    def update_azure_service_mesh_profile(self) -> ServiceMeshProfile:
```

Comment:
When reviewing the changes to `aks approuting` recently, I realized that it is similar to the implementation of the `aks mesh` command group. If function `_aks_mesh_update` (in `custom.py`) only wants to process several options supported by the `aks mesh` command group through the decorator framework and keep the values of other properties in the existing mc, I would recommend you replace `mc = aks_update_decorator.update_mc_profile_default()` with `mc = aks_update_decorator.fetch_mc()` to avoid unexpected behaviors caused by other update_xxx functions.